core-utilities/src/main/java/edu/illinois/cs/cogcomp/annotation/BasicAnnotatorService.java

```diff
@@ -371,7 +371,12 @@ public TextAnnotation addViewsAndCache(TextAnnotation ta, Set<String> viewsToAnn
                 ta = annotationCache.getTextAnnotation(ta);
 
         for (String viewName : viewsToAnnotate) {
-            isUpdated = addView(ta, viewName) || isUpdated;
+            try {
+                isUpdated = addView(ta, viewName) || isUpdated;
+            } catch (AnnotatorException e) {
+                // the exception is handled here, because one single view failure should not resutl in loss of all the annotations
```

Comment:
I agree that this should be configurable. But I think default should be "don't fail". 
Regarding downstream caching: I think the current approach works (text hash is used just to retrieve existing annotations; individual views are checked each time), though perhaps not for sentence-by-sentence.